scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+=begin
+  Documentation: TBD
+=end
+
+custom_require.call(%w(common common-items drinfomon))
+
+# Define Scroll Info class
+class ScrollInfo
+  def initialize(container, index, container_index, ordinal, spell, type)
+	# Instance variables
+	@container = container
+	@index = index
+	@container_index = container_index
+	@ordinal = ordinal
+	@spell = spell
+	@type = type
+  end
+
+  def container
+	return @container
+  end
+
+  def spell
+	return @spell
+  end
+
+  def ordinal
+	return @ordinal
+  end
+  
+  def index
+	return @index
+  end
+  
+  # represents the local index of the
+  # scroll within the container
+  def container_index
+	return @container_index
+  end
+
+  def get_type
+	return @type
+  end
+
+  def set_container(container)
+	@container = container
+  end
+  
+  def set_container_index(index)
+	@container_index = index
+  end
+  
+  def set_ordinal(ordinal)
+	@ordinal = ordinal
+  end
+end
+
+class ScrollSearch
+  include DRC
+  include DRCI
+
+  def initialize
+
+	# Variables
+	@scroll_count = 0
+	@global_scroll_index = UserVars.scroll_counter.nil? ? 0 : UserVars.scroll_counter
+	@owned_scrolls = UserVars.owned_scrolls.empty? ? {} : UserVars.owned_scrolls
+    
+	# RESET ALL
+	# UserVars.owned_scrolls = { }
+	# @owned_scrolls = { }
+	# UserVars.scroll_counter = 0
+
+	$settings = get_settings()
+	$debug_mode = $settings.debug_mode
+
+	echo("Settings debug mode is: #{$debug_mode}") if $debug_mode
+	if $settings.nil? || $settings.empty?
+	  echo("Settings was nil!") if $debug_mode
+	end
+
+	arg_definitions = 
+	[
+	  [
+		{ name: 'search', regex: /^sea.*/i, description: 'Searches given container and labels any unlabeled scrolls. ' },
+		{ name: 'container', regex: /\w+/i, description: 'The container to search. ' }
+	  ],
+	  [
+		{ name: 'list', regex: /^lis.*/i, optional: false, description: 'Lists all of the scrolls that have been categorized after a SEARCH. ' },
+	  ]
+	]
+
+	args = parse_args(arg_definitions)
+
+	unless args.search || args.list
+	  display_args(arg_definitions)
+	  return
+	end
+
+	pause
+
+	if args.search
+	  if args.container.nil?
+		echo_message("SPECIFY THE CONTAINER TO SEARCH!")
+		return
+	  else
```

Comment:
Because of the return one line 105, you don't need the else case here (i.e. if the return happens, line 107 won't happen even if the else is removed).